python/rtypes/dataframe/trigger_manager.py

```diff
@@ -202,12 +237,12 @@ def __procedure_index_location(self, trigger_obj):
         # changing the __eq__ method
 
         for index in range((len(
```

Comment:
Super inefficient. Since the list is already "sorted" use `bisect.bisect` to find the index instead. Much faster. (read binary search in CLR to understand how it works)